pkg/sanity/groupcontroller.go

```diff
@@ -0,0 +1,341 @@
+					},
+				},
+			}
+			_, err := r.CreateVolume(context.Background(), vol2Req)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should fail when the volume source snapshot is not found", func() {
+			if !isControllerCapabilitySupported(r, csi.ControllerServiceCapability_RPC_CREATE_DELETE_SNAPSHOT) {
+				Skip("Snapshot not supported")
+			}
+
+			By("creating a volume from source snapshot")
+			volName := UniqueString("sanity-controller-vol-from-snap")
+			volReq := MakeCreateVolumeReq(sc, volName)
+			volReq.VolumeContentSource = &csi.VolumeContentSource{
+				Type: &csi.VolumeContentSource_Snapshot{
+					Snapshot: &csi.VolumeContentSource_SnapshotSource{
+						SnapshotId: "non-existing-snapshot-id",
+					},
+				},
+			}
+			_, err := r.CreateVolume(context.Background(), volReq)
+			Expect(err).To(HaveOccurred())
+			serverError, ok := status.FromError(err)
+			Expect(ok).To(BeTrue())
+			Expect(serverError.Code()).To(Equal(codes.NotFound), "unexpected error: %s", serverError.Message())
+		})
+		*/
+	})
+
+	Describe("GetVolumeGroupSnapshot", func() {
+		It("should fail when no volume id is provided", func() {
+			_, err := r.GetVolumeGroupSnapshot(
+				context.Background(),
+				&csi.GetVolumeGroupSnapshotRequest{
+					Secrets: sc.Secrets.ListSnapshotsSecret,
+				},
+			)
+			Expect(err).To(HaveOccurred())
+
+			serverError, ok := status.FromError(err)
+			Expect(ok).To(BeTrue())
+			Expect(serverError.Code()).To(Equal(codes.InvalidArgument), "unexpected error: %s", serverError.Message())
+		})
+
+		It("should fail when an invalid volume id is used", func() {
+			_, err := r.GetVolumeGroupSnapshot(
+				context.Background(),
+				&csi.GetVolumeGroupSnapshotRequest{
+					GroupSnapshotId: sc.Config.IDGen.GenerateInvalidVolumeID(),
+					Secrets:         sc.Secrets.ListSnapshotsSecret,
+				},
+			)
+			Expect(err).To(HaveOccurred())
+
+			serverError, ok := status.FromError(err)
+			Expect(ok).To(BeTrue())
+			Expect(serverError.Code()).To(Equal(codes.NotFound), "unexpected error: %s", serverError.Message())
+		})
+	})
+
+	Describe("DeleteVolumeGroupSnapshot", func() {
+		It("should fail when no volume id is provided", func() {
+			_, err := r.DeleteVolumeGroupSnapshot(
+				context.Background(),
+				&csi.DeleteVolumeGroupSnapshotRequest{
+					Secrets: sc.Secrets.DeleteSnapshotSecret,
+				},
+			)
+			Expect(err).To(HaveOccurred())
+
+			serverError, ok := status.FromError(err)
+			Expect(ok).To(BeTrue())
+			Expect(serverError.Code()).To(Equal(codes.InvalidArgument), "unexpected error: %s", serverError.Message())
+		})
+
+		It("should succeed when an invalid volume id is used", func() {
+			_, err := r.DeleteVolumeGroupSnapshot(
+				context.Background(),
+				&csi.DeleteVolumeGroupSnapshotRequest{
+					GroupSnapshotId: sc.Config.IDGen.GenerateInvalidVolumeID(),
+					Secrets:         sc.Secrets.DeleteSnapshotSecret,
+				},
+			)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		/* disabled for now
```

Comment:
Are you planning to add more tests in this PR or in followup PRs?

Reply:
Right, I plan to add more tests in different PRs. I did not intend to have the commented out code included in this commit. It should be gone now.